firrtl/src/main/scala/firrtl/Emitter.scala

```diff
@@ -30,7 +31,7 @@ trait EmittedAnnotation[T <: EmittedComponent] extends NoTargetAnnotation with C
 }
 
 sealed trait EmittedCircuitAnnotation[T <: EmittedCircuit] extends EmittedAnnotation[T] {
-  override def getBytes = value.value.getBytes
+  override def getBytes: scala.collection.mutable.ArraySeq.ofByte = value.value.getBytes
```

Comment:
This matters a lot less here since we've deprecated the entire `import firrtl`, but this is going to be a common theme as we add `-Xsource:3` so I want to describe what we should do in cases like these.

We don't want the locally inferred type. In this case, we want the type from `CustomFileEmission.getBytes` which is `Iterable[Byte]`. I'm a little surprised you need a type annotation here though since my understanding is that `-Xsource-features:infer-override` would do what we want (making the type `Iterable[Byte]`). Maybe you made this change before adding that option?

Regardless, we do not want this type, same for similar things below. We also need to report types that have changed in release notes. I don't think we need to do it for this compilation unit (since `import firrtl._`) is deprecated anyway, but we will need to do so for `core` and `chisel` since it could affect users in some rare cases. I think we can do this by enabling `-Xsource:3` _without_ `-Xsource-features:infer-override`, collecting the errors for release notes, then adding `-Xsource-features:infer-override`. We will need to make sure we don't hit the max number of errors when doing that.

Reply:
hm not sure i understand - all type annotation additions in commit 2dfb9d0 are quickfixable compiler rewrites

Reply:
that is i did not add these manually, they were added by enabling infer-overrides

Reply:
My point is that the quickfix rewrite is not what we want. We want the type that Scala 3 will infer which is what the error is about. That type in this case is `Iterable[Byte]`. This is a case where we want to change Chisel's public API a bit in prep for the Scala 3 upgrade.